gtest/src/log.rs

```diff
@@ -396,6 +396,40 @@ impl RunResult {
             .flat_map(DecodedCoreLog::try_from_log)
             .collect()
     }
+
+    /// Trying to get the panic log.
+    fn panic_log(&self) -> Option<&CoreLog> {
+        let [core_log] = &self.log[..] else {
+            return None;
+        };
+        let is_panic = matches!(
+            core_log.reply_code(),
+            Some(ReplyCode::Error(ErrorReplyReason::Execution(
+                SimpleExecutionError::UserspacePanic
+            )))
+        );
+        is_panic.then_some(core_log)
+    }
+
+    /// If the main message panicked.
+    pub fn panicked(&self) -> bool {
```

Comment:
redundant, you have `self.main_failed() -> bool` doing the same

Reply:
But isn't main failed not just panics? It could be OutOfGas or StackOverflow or any other error that could make the message fail?